[Howlong11] iP

Howlong11 opened the pull request:
# LEMONA

Feeling **Exhausted** :weary:? Start planning your future with _LEMONA_, your plan assistance and the Best source for Vitamin!

> "This is not sponsored by Lemona" - Somebody who made this program

## LEMONA can help you by providing you sufficient amount of the following vitamins!
1. Vitamin C
2. Vitamin B2
3. Vitamin B6

## Also, LEMONA can help you planning your future like:
- Task management
- Deadline tracking
- Event scheduling

## Have you had enough vitamins?
- [x] Yes! I had LEMONA!
- [x] I think so? I had some fruits!
- [x] No, I will go take some LEMONA


You can run Lemona with the `main` method of Duke class!
```java
public class Duke {
    public static void main(String[] args) {
        Lemona lemona = new Lemona("data/lemona.txt");
        lemona.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
FIrst commit, level-0

Commit:
second commit. Level-1

Commit:
Revert "second commit. Level-1"

This reverts commit 8cc892402b5567c506b3b43e3bde76b3b2b1ab23.

Commit:
Second commit, Level-1

Commit:
Third commit, Level-2

Commit:
Fourth commit, level-3

Commit:
Fifth commit, level-4

Commit:
Sixth commit, A-textUItesting

Commit:
Seventh commit, level-5

Commit:
eighth commit, level-6

Commit:
ninth commit, error fixes

Commit:
Save list into txt file, level-7

Commit:
Merge branch 'branch-Level-7'

Commit:
Save dates as LocalDateTime objects, level-8

Commit:
Merge branch 'branch-Level-8'

Commit:
OOP structure, A-MoreOOP

Commit:
data file

Commit:
Merge branch 'branch-MoreOOP'

Commit:
create packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Set up gradle

Commit:
add junit tests

Commit:
commit for jar

Commit:
Javadoc

Commit:
coding standard

Commit:
add Find method

Jawad280 commented on src/main/java/Lemona.java:
```diff
@@ -0,0 +1,93 @@
+import task.Event;
+import task.Deadline;
+
+/**
```
I like the detailed JavaDocs for the class 👍

Jawad280 commented on src/main/java/Lemona.java:
```diff
@@ -0,0 +1,93 @@
+            switch (parsedInput[0]) {
+            case ("bye"):
+                ui.bye();
+                System.exit(0);
+                break;
+            case ("mark"):
+                tasks.mark(Integer.parseInt(parsedInput[1]));
+                storage.save(tasks);
+                break;
+            case ("unmark"):
+                tasks.unmark(Integer.parseInt(parsedInput[1]));
+                storage.save(tasks);
+                break;
+            case ("delete"):
+                tasks.delete(Integer.parseInt(parsedInput[1]));
+                storage.save(tasks);
+                break;
+            case ("list"):
+                ui.list(tasks);
+                break;
+            case ("todo"):
+                Task task = new Todo(parsedInput[1]);
+                tasks.add(task);
+                storage.save(tasks);
+                break;
+            case ("deadline"):
+                task = new Deadline(parsedInput[1], parsedInput[2]);
+                tasks.add(task);
+                storage.save(tasks);
+                break;
+            case ("event"):
+                task = new Event(parsedInput[1], parsedInput[2], parsedInput[3]);
+                tasks.add(task);
+                storage.save(tasks);
+                break;
+            case ("find"):
+                tasks.find(parsedInput[1]);
+                break;
+            default:
+                break;
```
Perhaps you can add these condition checks in the Parser class ?

Jawad280 commented on src/main/java/Task/Deadline.java:
```diff
@@ -0,0 +1,61 @@
+            str = "[D]" + super.print() + "(by: " + time + ")";
+        } catch (DateTimeParseException e) {
+            System.out.println(LINE);
+            System.out.println("\t I think you haven't had enough vitamin C."
+                    + "\n\t Your time format should be :"
+                    + "\n\t\t { dd/MM/yyyy HHmm }"
+                    + "\n\t I suggest you take some LEMONA.");
```
I like the custom error messages 👍

purivirakarin replied:
Agree, nice custom error message. Just a suggestion, this message may be declared as a constant.

Jawad280 commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,74 @@
+ * Represents a generic task in the task manager application.
+ * A task has a description and can be marked as done or undone.
+ */
+public class Task {
```
Perhaps Task can be an abstract class ?

Jawad280 reviewed (commented):
Overall, I found your code easy to read for most places. I noted some minor changes regarding the Parser class and handling the different cases. Other than that it generally looks good and also follows coding standards !

purivirakarin commented on src/main/java/Task/Event.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Override
+    public String getTaskInfo() {
+        String start = startTime.format(outputFormat);
+        String end = endTime.format(outputFormat);
```
For more descriptive variable name, you may rename `start` and `end` into `formattedStartTime` and `formattedEndTime`.

purivirakarin commented on src/main/java/Lemona.java:
```diff
@@ -0,0 +1,93 @@
+        ui.greet();
+        while (true) {
+            String input = ui.readInput();
+            String[] parsedInput = Parser.parse(input);
```
According to the coding convention, you may rename `parsedInput` into `parsedInputs` (plural form), as it is a collection of Strings.

purivirakarin reviewed (commented):
The code quality is good and has descriptive comments/JavaDocs. Just some points I have mentioned about the variables naming (plural form) and some reusable texts may be declared as constants :)

Commit:
add styletest using gradle

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Create a GUI of Lemona chatbot

Commit:
Merge branch 'branch-Level-10'

Commit:
Add assertions

TaskList and Parser classes have no assertions.

These classes can cause errors if wrong input is entered.

Adding assertions allow to prevent possible bugs.

Assertions are added instead of exceptions as they could be prevented if original code was written in better ways.

Commit:
Improve code quality of Parser class

parse() method of Parser class was too long

part of parse() method is extracted

parseDates() method is extracted from parse() method to make it shorter

Howlong11 committed:
Merge pull request #2 from Howlong11/branch-A-Assertions

Add assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/lemona/oop/Parser.java

Howlong11 committed:
Merge pull request #3 from Howlong11/branch-A-CodeQuality

Improve code quality of Parser class

Commit:
detect duplicate tasks added

Commit:
add title to GUI

Commit:
add GUI screenshot

Howlong11 committed:
Update README.md

Howlong11 committed:
Update README.md

Commit:
Fix Junit unit tests

Commit:
Merge branch 'master' of https://github.com/Howlong11/ip

Commit:
Fix Junit unit testing

Commit:
renamed Ui.png

Howlong11 committed:
Rename Ui.png.png to Ui.png

Howlong11 committed:
Update README.md
